CMakeLists.txt

```diff
@@ -16,6 +16,7 @@ else()
 endif()
 
 # --- compiler flags
+set(CMAKE_CXX_STANDARD 11)
```

Comment:
It is my understanding that `CMAKE_CXX_STANDARD`
is more targeted to toolchain files than for the projects directly:

- https://gitlab.kitware.com/cmake/cmake/issues/17146#note_299405

Whereas `target_compile_features(libninja PUBLIC cxx_std_11)`,
just specifies a minimum required standard,
but allow a developer or package to upgrade the standard seamlessly,
e.g. a packager may want to upgrade to C++20 for ABI-compatibility reasons.
This can be done using `CMAKE_CXX_STANDARD` from a toolchain file,
or from the command line:

    cmake -DCMAKE_CXX_STANDARD=20 ...

CMake will also respect the `-std=<standard>` flag,
if specified specified from the environment variable `CXXFLAGS`:

    env CXXFLAGS=-std=c++20 cmake ...

or from the command line `CMAKE_CXX_FLAGS`:

    cmake -DCMAKE_CXX_FLAGS=-std=c++20 ...

One thing to be aware with this method,
is that if one wants to make sure, e.g. in CI,
that the minimum C++ standard is actually tested,
it should do so explicitly, using one of the aforementioned methods.